core/sql/optimizer/ControlDB.cpp

```diff
@@ -108,7 +108,12 @@ ControlDB::~ControlDB()
 
 void ControlDB::setRequiredShape(ControlQueryShape *shape)
 {
-  delete requiredShape_;
+  if (requiredShape_)
+    {
+      delete requiredShape_->getShape();
+      delete requiredShape_;
```

Comment:
Yes, one thing I don't like about NABasicObject is that the heap pointer is junk for NABasicObjects that are allocated on the stack. Fortunately that's not a problem here. The allocation/deallocation of objects is pretty clear in this file. Thanks again for finding this leak!